sensor/kubernetes/sensor/sensor.go

```diff
@@ -57,7 +57,7 @@ var (
 )
 
 // CreateSensor takes in a client interface and returns a sensor instantiation
-func CreateSensor(client client.Interface, workloadHandler *fake.WorkloadManager) (*sensor.Sensor, error) {
+func CreateSensor(client client.Interface, workloadHandler *fake.WorkloadManager, connFactory connection.GRPCConnectionFactory) (*sensor.Sensor, error) {
```

Comment:
My idea was that the `GRPCConnectionFactory` is **not** optional. When running on a real cluster, `kubernetes/main.go` code will inject a real factory in Sensor. 

`CreateSensor` is also not far down the stack, it's called on `kubernetes/main.go`. Maybe what we could do is to provide `opts...` param for both `conn` and `workloadHandler`, which were both created for non-prod requirements. 

Calling `CreateSensor` would look something like:
```golang
CreateSensor(k8sClient, WithWorkload(workloadManager), WithCentralConnectionFactory(connFactory))
```

WDYT?

Reply:
Actually, for `workloadHandler` it would require further refactor to pass it as an optional param, because it isn't just setting some property in sensor. Instead, it registers signal handlers to some sensor's components:

```golang
if workloadHandler != nil {
       workloadHandler.SetSignalHandlers(processPipeline, networkFlowManager)
}
```

I like the idea of passing conditional arguments here, I was already thinking about how to do that. But I think we need to wait until more parameters need to be passed here. We might need to add a `localSensor` property as my draft code in #1730, plus there are plans to make changes in the fake workload. So maybe once we make further changes here we can figure out what's the right abstraction for the `opts...` parameter. 

Would you be ok with that?

Reply:
Ah okay, let's stick with your approach for now. 👍